### ISSUE TITLE
ContinuanceToken ToString always fails, so they cannot be logged out or seen

### ISSUE THREAD

**Issue by secondimpactrob** [PlayEveryWare/eos_plugin_for_unity#674]
The `ContinuanceToken` class, which is sometimes returned from login API calls, cannot be debug-logged out since its `ToString` functionality always fails with result `LimitExceeded`.

```c#
public Result ToString(out Utf8String outBuffer)
{
    int inOutBufferLength = 1024;
    System.IntPtr outBufferAddress = Helper.AddAllocation(inOutBufferLength);

    var funcResult = Bindings.EOS_ContinuanceToken_ToString(InnerHandle, outBufferAddress, ref inOutBufferLength);

    Helper.Get(outBufferAddress, out outBuffer);
    Helper.Dispose(ref outBufferAddress);

    return funcResult;
}
```

It seems this code only attempts `EOS_ContinuanceToken_ToString` with a buffer size of **1024** (which must be too small) and reports the result. Unfortunately, doesn't seem the innards are exposed in any way that would allow me to make the call myself to work around this.

According to a comment in the source, Epic suggest you can call `EOS_ContinuanceToken_ToString` and pass a null buffer and it will report the size required.

It's very useful to be able to log out and inspect the various ids and tokens coming and going, and in my case this discovery has come after a while spent assuming I was getting back 'no continuance token' from calls, since they were logging out as empty strings, when actually I _am_ getting valid continuance tokens.

**Comment by paulhazen** [comment]
> According to a comment in the source, Epic suggest you can call `EOS_ContinuanceToken_ToString` and pass a null buffer and it will report the size required.

This seems like a reasonable approach (to alter the aforementioned `ToString` method to first pass a `null` buffer). Are you able to confirm that making that change locally resolves your problem? If so, we can forward the feedback to Epic - as the `ContinuanceToken.cs` file is a part of the EOS SDK, and changes we might make to the source as it is included in the plugin will be overwritten when we upgrade to newer versions of the SDK. This means that (temporarily) the best solution may be to make that change locally - but that the long-term solution should really be that Epic corrects their implementation. Does that make sense?

Let me know if you are able to confirm that your proposed solution resolves your issue - and I'll include it in our list of things that we periodically bring to Epic in the form of feedback.
 
> It's very useful to be able to log out and inspect the various ids and tokens coming and going, and in my case this discovery has come after a while spent assuming I was getting back 'no continuance token' from calls, since they were logging out as empty strings, when actually I _am_ getting valid continuance tokens.

That's frustrating - and sounds like a familiar predicament (making assumptions in general). Sorry that has been your experience. I can see how logging those values would be critical to debugging.

**Comment by paulhazen** [comment]
This issue should be resolved now in `3.2.0` - although I would wait a day before upgrading - we have discovered some issues that need to be resolved with that release. I'm closing out this issue as the central problem has been addressed.
